bin/warehouse/csv_diff.py

```diff
@@ -0,0 +1,107 @@
+import json
+import requests
+import sys
+import csv
+import os
+import codecs
+from sys import stderr, stdout, argv
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+
+requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+
+def get_fieldnames(fields):
+    fieldnames = ",".join([f['id'] for f in fields])+",owner_org,log_date,log_activity"
+    return fieldnames
+
+def csv_to_dict(csv_file, primary_keys):
+    csv_dict = {}
+    with open(csv_file) as f:
+        hashcsv = csv.DictReader(codecs.EncodedFile(f, 'utf-8', 'utf-8-sig'), delimiter=",")
+        for row in hashcsv:
+            primary_fields = [str(row[t]) for t in primary_keys]
+            uid = '-'.join(primary_fields)
+            csv_dict[uid] = row
+    return csv_dict
+
+def compare_dicts(prev, curr):
+    removed_keys = []
+    added_keys = []
+    modified_keys = []
+    for key in prev:
+        if not key in curr:
+            removed_keys.append(key)
+    for key in curr:
+        if not key in prev:
+            added_keys.append(key)
+        else:
+            for field in curr[key]:
+                if curr[key][field] != prev[key][field]:
+                    modified_keys.append(key)
+                    break
+    return removed_keys, added_keys, modified_keys
+
+def add_metadata_fields(prev, curr, rkeys, akeys, mkeys, date):
+    results = {}
+    result_keys=[]
+    for key in akeys:
+        curr[key]["log_date"] = date
+        curr[key]["log_activity"] = 'C'
+        results[key] = curr[key]
+        result_keys.append(key)
+
+    for key in mkeys:
+        curr[key]["log_date"] = date
+        curr[key]["log_activity"] = 'M'
+        results[key] = curr[key]
+        result_keys.append(key)
+
+    for key in rkeys:
+        prev[key]["log_date"] = date
+        prev[key]["log_activity"] = 'D'
+        results[key] = prev[key]
+        result_keys.append(key)
+
+    return results, result_keys
+
+
+prev_csv, current_csv, endpoint, datestamp, outfile = sys.argv[1], sys.argv[2], sys.argv[3], sys.argv[4], sys.argv[5]
+
+field_info = requests.get(endpoint, timeout=100, verify=False).json()
+
+
+# Grab the primary key fields from the datatype reference endpoint
+if 'nil' in current_csv:
```

Comment:
the fact that our current two-page types always have a `'nil'` in their second page is a detail that shouldn't appear in this script. Match the name to the resource name instead so this doesn't fail on new types.

Reply:
This wasn't resolved either. @HoussamBedja it's up to you whether to push this to another ticket.